src/navigation/RootNavigator.tsx

```diff
@@ -13,7 +14,10 @@ function RootNavigator() {
         <Route path={RoutesList.Home} element={<HomePage />} />
         <Route path={RoutesList.Login} element={<LoginPage />} />
         <Route path={RoutesList.Register} element={<RegistrationPage />} />
-        <Route path={RoutesList.Dashboard} element={<DashboardPage />} />
+        <Route
+          path={RoutesList.Dashboard}
+          element={<ProtectedRoute outlet={<DashboardPage />} />}
```

Comment:
```
<ProtectedRoute>
  <DashboardPage />
</ProtectedRoute>
```

and then you dont need to make new interface

Reply:
I changed the protected route to something like auth wrapper. I mean this is something like a group of routes that must be authenticated.  I couldn't do children prop without any element or tag because React node can be null from children and the only way to fix this is to add an empty element, which is a bad way for you. So I decided to workaround this with auth wrapper and this is correct now I think.